### ISSUE TITLE
Version bump?

### ISSUE THREAD

**Issue by johnmyleswhite** [JuliaLang/atom-language-julia#7]
Assuming we're replacing the existing language-julia package, I think we need a higher version number. Right now, my (heavily modified local install of) Atom prompts me to upgrade this package to the old package that we're replacing.

**Comment by spencerlyon2** [comment]
Yes, we should definitely do this. 

@tpoisot is the maintainer of the current package and he said (in a comment on #2) that he would de-register the current on on apm this weekend. Once that happens the one here can take its place, probably starting at the next minor version from where the current one left off.

**Comment by tpoisot** [comment]
@spencerlyon2 / @johnmyleswhite -- thanks for taking care of this. I've unpublished the old package, you're free to go.

**Comment by johnmyleswhite** [comment]
Thank you for starting this work, @tpoisot.

It looks like we didn't need to bump version numbers; once your package was unpublished, version 0.0.0 became the latest version.

Since things are working now, perhaps we shouldn't bother increasing the version number again?

**Comment by tpoisot** [comment]
I'm fine with starting a 0.0.0. Leaves a lot of room for improvement :wink:

**Comment by spencerlyon2** [comment]
Sounds good. Just pushed v0.1.0 (atom updated when I published)

**Comment by sbromberger** [comment]
Just FYI - my Atom doesn't see this as a new version - when I go into my installed packages, I'm showing  language-julia installed at v0.1.0 with no updates available. Clicking on the repo link brings me to atom-language-julia, though.

What am I doing wrong?

**Comment by spencerlyon2** [comment]
Not sure...

If you uninstall then re-install which version do you get (should be version 0.1.1 now)?

**Comment by sbromberger** [comment]
Still showing "all packages up to date" within Atom. Let me uninstall and give that a try.

**Comment by spencerlyon2** [comment]
@sbromberger, so you are good with the install?

**Comment by sbromberger** [comment]
Looks like it. How do I tell beyond checking what Atom reports?

**Comment by spencerlyon2** [comment]
Probably check for a new feature. You could:

- Enter any regex (e.g. `r"(\w+)"`), put the cursor on it, and run the `log cursor at scope` command (command-option-p on OS X).
- Try to enter an emoji `\:ghost:[TAB]` and see if it works
- Go to the settings page for the package, then view code, then check to see that the code there matches what is in the repo here.

**Comment by sbromberger** [comment]
1) ```Scopes at Cursor

source.julia
string.regexp.julia
constant.character.escape.julia```

2) \:ghost:[TAB] shows an autocomplete option but doesn't actually do anything.

**Comment by spencerlyon2** [comment]
Did you make sure to do the slash `\` and surround `ghost` with colons?

**Comment by sbromberger** [comment]
yup: here's a screenshot:

![screen shot 2015-06-15 at 09 47 21](https://cloud.githubusercontent.com/assets/941359/8162538/8adbca8c-1343-11e5-8d44-61cc810d739f.png)

**Comment by spencerlyon2** [comment]
Yeah it does. Strange. I don't see that on my end (in fact the autocomplete-plus tooltip doesn't pop up, even if I try to force it with ctrl-space).

**Comment by spencerlyon2** [comment]
Confirmed, if I try to do this in a session where I have activated Hydrogen, the snippet here gets clobbered by the same list you show in your screen shot. Not sure how Hydrogen is causing this.

cc: @willwhitney

**Comment by sbromberger** [comment]
Confirmed: when Hydrogen is disabled, ghost works.

**Comment by sbromberger** [comment]
I need Hydrodgen more than I need emoji, so it's not a deal-breaker :)

**Comment by willwhitney** [comment]
I think this is happening because of how Hydrogen's prefix regex for `autocomplete-plus` is written. I was worried something like this would happen...

The regex Hydrogen uses to find the prefix for autocomplete is `/([\w0-9_-][\.:\$]?)+$/`. This treats colons as internal to a symbol to work with languages like Lua (where methods are called with `object:method()`). That means that it's finding the symbol before the mark as `ghost:`, which is a long enough symbol to ask for completion from the Julia kernel, which then returns results for a zero-length prefix because `:` isn't legal inside variable names in Julia.

Long story short, I'm going to need to change to using custom regexes for every language. Sorry for the trouble!

**Comment by sbromberger** [comment]
so, I updated Hydrogen to 0.4.0. Here's what I found:

The first use of `\:ghost:[TAB]` succeeds in creating the emoji. However, once I activate hydrogen, it no longer works. See screenshot:

![screen shot 2015-06-15 at 17 54 13](https://cloud.githubusercontent.com/assets/941359/8172576/94f295e8-1387-11e5-9361-14facfb5d9fa.png)

**Comment by willwhitney** [comment]
Yes, 0.4.0 doesn't yet include a fix for this. Should be forthcoming in the
next day or so though.

**Comment by sbromberger** [comment]
@willwhitney ah, apologies. I saw a new version number and assumed that the fix had been applied... thanks for all your help!

**Comment by willwhitney** [comment]
Does anyone know where I can find a regex for valid Julia identifiers? With high unicode characters and whatnot allowed, it seems quite a bit more complicated than most languages.
